### DISCUSSION TITLE
Request: add ability to add tables for specific fields in Diagram View

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#1106] [Ideas]
It would be nice if I could right-click on a field in a table in diagram view and only add tables that filter _that field_ vs tables that filter the entire table, or all tables that that table filters. Just more granular, and saves me from having to go to the relationship view to figure out what tables(s) I want to add to a diagram.

**Comment by otykier** [comment]
What do you mean when you say “tables that filter that field”? A relationship between two tables always causes the entire table at the end of the filter direction to become filtered - not just the column participating in the relationship.

**Comment by edhans** [comment]
Use the Contoso model as an example, and Internet Sales is my fact table. I am working on a measure that need to come through the ProductKey field. I want to create a simplified diagram by

1. Drag in Internet Sales
2. Right-click on ProductKey
3. Select "Add tables that filter this field"
4. Product table is added.

My terminology probably isn't right. By "filter this field" I mean "show me other tables that are filtering this table based on a relationship with this field" - but that is a very large right-click item. 😁

This is very helpful when the table names and field names aren't logical. We've all worked with models were the Product table is called P100A and the field names are equally nonsensical.

**Comment by otykier** [comment]
Gotcha. So something like “Add tables related to this column” (it’s rare to have more than one table related to a fact table column, though, but I understand why it’s sometimes useful to include only specific related tables in the diagram, rather than all related tables).

We’re also working on some much needed visual updates on the diagram. For example, we’ll have small icons on every column that indicates if a column participates in a relationship, if it’s a calculated column, etc…

**Comment by edhans** [comment]
Exactly - just something more granular than "all tables filtering this table."

Looking forward to the updates for the Diagram view. Those will help when trying to figure out what is going on in a model you've inherited. Thanks!

**Comment by edhans** [comment]
Just pinging here to ensure now that I think my request has been better explained, it is no longer awaiting reply but an enhancement request.
